### PR TITLE
Migration to Gradle Kotlin DSL

### PR DESCRIPTION
**Pull request by danielpetisme**
A proposal for #73 

@jponge could you have a look 🙏

### CODE REVIEW

**Comment by jponge** [comment]
I will definitely review 👍

Le mar. 12 mars 2019 à 23:36, Thomas Segismont <notifications@github.com> a
écrit :

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.gradle.api.tasks.wrapper.Wrapper.DistributionType.ALL
+
+plugins {
+  java
+  id("io.vertx.vertx-plugin") version "0.4.0"
```
Comment (tsegismont) [inline build.gradle.kts]: This version changes the default uber jar classifier from `fat` to `all`. You need to either update all the deployment scripts or configure the classifier. My preference goes to the latter :wink:

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+repositories {
+  jcenter()
+  mavenLocal()
+}
+
+group = "io.vertx"
+version = "0.5.0-SNAPSHOT"
```
Comment (tsegismont) [inline build.gradle.kts]: Should be `2.0.1`

File: build.gradle.kts (hunk 1 above)
Comment (jponge) [reply build.gradle.kts]: Yep, embrace the new!

**Comment by jponge** [comment]
Just one thing: you may extract `val vertxVersion = "..."` then use `$vertxVersion` in all dependencies.

**Comment by danielpetisme** [comment]
I've updated the PR.
the only concern I have is about the classifier.
I don't understand why but the following snippet does not work.
```
tasks.withType<ShadowJar> {
  archiveClassifier.set("fat")
}
```

I've ended with a more brutal solution...
```
tasks.withType<ShadowJar> {
  archiveFileName.set("${project.name}-${project.version}-fat.jar")
}
```

WDYT?

**Comment by jponge** [comment]
With the latest version of the plugin it should be `-all.jar`.

**Comment by tsegismont** [comment]
Daniel, Julien, the starter is back online and running on HTTPS. As soon as
we can merge this one I'll release 2.0.1.

**Comment by danielpetisme** [comment]
I first went to Thomas's proposal of only updating the classifier ans keep the script. This morning I understood the "Embrace the new" reply 😄 

@tsegismont I've update the deploy script but didn't test it on a dev environment... 🤞

**Comment by tsegismont** [comment]
@danielpetisme look at:

https://github.com/vert-x3/vertx-starter/blob/3ea4b36fc9b6fd6a0cd397f879f0d3354766547a/scripts/deploy.sh#L18

Plus the service script on the machine works assuming the thing is named _fat_ something.

I really prefer to play this safe and keep the same output name.

**Comment by danielpetisme** [comment]
So basically I rolback to
https://github.com/vert-x3/vertx-starter/pull/79/commits/75eef1be0f4d9e40b0aee0ef897be845303bb6e8

**Comment by tsegismont** [comment]
I guess so. Can you also rebase on master (and set version to 2.0.2). Thanks

**Comment by danielpetisme** [comment]
Done

**Comment by tsegismont** [comment]
@danielpetisme thanks!

**Pull request merged**